File: metatag/utils/string_utils.py
"""String utilities."""


def to_title_case(s: str) -> str:
    """Convert string to Title Case."""
    if not s:
        return s
    words = s.split()
    title_words = []
    for word in words:
        if word:
            title_words.append(word[0].upper() + word[1:].lower())
    return " ".join(title_words)


def to_sentence_case(s: str) -> str:
    """Convert string to Sentence case."""
    if not s:
        return s
    return s[0].upper() + s[1:]


def to_upper(s: str) -> str:
    """Convert string to uppercase."""
    return s.upper()


def to_lower(s: str) -> str:
    """Convert string to lowercase."""
    return s.lower()

File: src/utils/string_utils.cpp
#include "string_utils.h"
#include <QStringList>
#include <QRegularExpression>

namespace metatag {

QString StringUtils::toTitleCase(const QString& str)
{
    if (str.isEmpty())
        return str;
    
    QStringList words = str.split(' ', Qt::SkipEmptyParts);
    for (auto& word : words) {
        if (!word.isEmpty()) {
            word = word.front().toUpper() + word.mid(1).toLower();
        }
    }
    return words.join(' ');
}

QString StringUtils::toSentenceCase(const QString& str)
{
    if (str.isEmpty())
        return str;
    
    QString result = str;
    result[0] = result[0].toUpper();
    return result;
}

QString StringUtils::toUpper(const QString& str)
{
    return str.toUpper();
}

QString StringUtils::toLower(const QString& str)
{
    return str.toLower();
}

}